client/me/notification-settings/reader-subscriptions/index.jsx

```diff
@@ -25,270 +25,259 @@ import { recordGoogleEvent } from 'calypso/state/analytics/actions';
 import { getReaderTeams } from 'calypso/state/teams/selectors';
 import SubscriptionManagementBackButton from '../subscription-management-back-button';
 
-class NotificationSubscriptions extends Component {
-	handleClickEvent( action ) {
-		return () => this.props.recordGoogleEvent( 'Me', 'Clicked on ' + action );
-	}
+const NotificationSubscriptions = ( props ) => {
+	const dispatch = useDispatch();
+	const teams = useSelector( getReaderTeams );
+	const isAutomattician = isAutomatticTeamMember( teams );
+	const { translate } = props;
 
-	handleFocusEvent( action ) {
-		return () => this.props.recordGoogleEvent( 'Me', 'Focused on ' + action );
-	}
+	const handleClickEvent = useCallback(
+		( action ) => {
+			return () => dispatch( recordGoogleEvent( 'Me', 'Clicked on ' + action ) );
+		},
+		[ dispatch ]
+	);
 
-	handleCheckboxEvent( action ) {
-		return ( event ) => {
-			const eventAction = 'Clicked ' + action + ' checkbox';
-			const optionValue = event.target.checked ? 1 : 0;
+	const handleFocusEvent = useCallback(
+		( action ) => {
+			return () => dispatch( recordGoogleEvent( 'Me', 'Focused on ' + action ) );
+		},
+		[ dispatch ]
+	);
 
-			this.props.recordGoogleEvent( 'Me', eventAction, 'checked', optionValue );
-		};
-	}
+	const handleCheckboxEvent = useCallback(
+		( action ) => {
+			return ( event ) => {
+				const eventAction = 'Clicked ' + action + ' checkbox';
+				const optionValue = event.target.checked ? 1 : 0;
 
-	getDeliveryHourLabel( hour ) {
-		return this.props.translate( '%(fromHour)s - %(toHour)s', {
-			context: 'Hour range between which subscriptions are delivered',
-			args: {
-				fromHour: this.props.moment().hour( hour ).minute( 0 ).format( 'LT' ),
-				toHour: this.props
-					.moment()
-					.hour( hour + 2 )
-					.minute( 0 )
-					.format( 'LT' ),
-			},
-		} );
-	}
+				dispatch( recordGoogleEvent( 'Me', eventAction, 'checked', optionValue ) );
+			};
+		},
+		[ dispatch ]
+	);
 
-	render() {
-		const { locale, teams } = this.props;
-		const isAutomattician = isAutomatticTeamMember( teams );
+	const getDeliveryHourLabel = useCallback(
+		( hour ) => {
+			return translate( '%(fromHour)s - %(toHour)s', {
+				context: 'Hour range between which subscriptions are delivered',
+				args: {
+					fromHour: props.moment().hour( hour ).minute( 0 ).format( 'LT' ),
+					toHour: props
+						.moment()
+						.hour( hour + 2 )
+						.minute( 0 )
+						.format( 'LT' ),
+				},
+			} );
+		},
+		[ props ]
```

Comment:
Also, note that usually when we provide all props as memo dependencies, it's a sign that the memo is likely redundant.